backstop_scenarios.js

```diff
@@ -78,22 +82,28 @@ try {
   scenarioPaths = defaultPaths.paths;
 }
 
-/**
- * @todo we need to allow for a deep merge so that the template-paths.js can include an override for the Home and
- * Forced 404 paths.
- */
-for (let i=0;i<scenarioPaths.length;++i) {
-  let newScenario = scenarioPaths[i];
-  let newPath = newScenario.path;
-  scenarioPaths[i].referenceUrl = `${baselineURL}${newPath}`;
-  scenarioPaths[i].url = `${testURL}${newPath}`;
+//now that we have our initial collection of paths, we need to merge any that are duplicates
+newScenarios = scenarioPaths.reduce((accumulator, currentValue, currentIndex, array) => {
+  //have we already processed a path with this label?
+  let alreadyExists = Object.keys(accumulator).filter(k=>accumulator[k]['label'] === currentValue['label']);
```

Comment:
Can `Array.find` be useful here as you only want the first entry if it exists?

```
let alreadyExisting = Object.keys(accumulator).find(entry => entry['label'] === currentValue['label']);
```

Reply:
What I'm trying to do here is locate the array _key_ for an entry in the accumulator array where an object's label in the accumulator array matches the label of the `currentValue.label`. 

Doing a `find` on `Object.keys(accumulator)` won't work since find gives us an array of the accumulators keys. And i dont _think_ a `find` on accumulator will work because we need the matching key and not the item from the array itself.

Reply:
HOWEVER, your comment got me thinking... We can do this though!
```
let alreadyExists = accumulator.findIndex(item => item.label === currentValue.label);
```